### PR TITLE
Basic structure for tabbed user settings

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/riot-web/issues/8199
This is part 1 of https://github.com/vector-im/riot-web/issues/7605

Opening this PR for review now in the interest of trying to keep these relatively small. 

In this PR:
* Modal tweaks for user settings
* `TabbedView` (inherited from https://github.com/matrix-org/matrix-react-sdk/pull/1644 and co.)
* Placeholder views for not-yet-implemented tabs
* A very early proof of concept for profile changes (trying to keep the hard stuff out of this PR - will be bundled into the next PR)
* A button to visit the old settings. This is expected to survive for about a week before the settings are completely usable for everyone. At this stage, people are probably going to be pressing it all the time however incremental improvements over the next week should make that less and less likely.

**Demo**:
settings-demo

**Better quality screenshot**:
image

### CODE REVIEW

File: res/css/structures/_TabbedView.scss [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  position: relative;
+}
+
+.mx_TabbedView_tabLabel_icon > .mx_TabbedView_maskedIcon {
+  width: 14px;
+  height: 14px;
+  display: inline-block;
+}
+
+.mx_TabbedView_tabLabel_icon > .mx_TabbedView_maskedIcon:before {
```
Comment (turt2live) [inline res/css/structures/_TabbedView.scss]: @bwindels this is how I ended up doing masking for the tabs. The basic structure is listed here to make them fit correctly, and the actual mask is defined elsewhere (eg `.mx_UserSettingsDialog_settingsIcon:before`). The `Tab` constructor takes an element (a `span` with the classname for the mask) and that gets put through to the DOM like so: 

image

end result is dynamic tab icons which also magically get masked when the active tab changes.

File: .eslintrc.js [hunk 2]
```diff
@@ -42,9 +42,8 @@ module.exports = {
 
         // bind or arrow function in props causes performance issues
         // (but we currently use them in some places)
-        "react/jsx-no-bind": ["warn", {
-            "ignoreRefs": true,
-        }],
+        // It's disabled here, but we should using it sparingly.
+        "react/jsx-no-bind": "off",
```
Comment (turt2live) [inline .eslintrc.js]: I've disabled this expecting that reviewers will complain loudly if the usage seems wrong, and to make the linter actually happy with the changeset in this PR.

File: res/css/structures/_TabbedView.scss (hunk 1 above)
Comment (turt2live) [reply res/css/structures/_TabbedView.scss]: The label is a flexbox and that seems to cause problems with the mask, so I ended up shoving the whole thing one level deeper.

File: .eslintrc.js (hunk 2 above)
Comment (turt2live) [reply .eslintrc.js]: We use it in several other places as well. I didn't seem to mention it, but that comment applies to the future as well.

File: res/css/structures/_TabbedView.scss (hunk 1 above)
Comment (turt2live) [reply res/css/structures/_TabbedView.scss]: Something to do with the alignment last I tried. I'll give it another go though.

Comment (turt2live) [reply res/css/structures/_TabbedView.scss]: right, so this happened:
image

I've managed to fix it though - please take another look at this / https://github.com/matrix-org/matrix-react-sdk/pull/2476/commits/0deb210fd89b91b41d147c4a08969ea9acc1b480

Comment (bwindels) [reply res/css/structures/_TabbedView.scss]: mask-repeat: no-repeat will fix that

**Comment by turt2live** [comment]
alright, @bwindels, this should finally be ready for review again.

**Pull request merged**